proxy/src/main/java/com/wavefront/agent/listeners/otlp/OtlpMetricsUtils.java

```diff
@@ -585,29 +559,115 @@ static void appendPositiveBucketsAndExplicitBounds(
     bucketCounts.add(0L);
   }
 
-  private static class BucketHistogramDataPoint {
+  static class CumulativeBucket {
+    private final String tag;
+    private final long count;
+
+    CumulativeBucket(String tag, long count) {
+      this.tag = tag;
+      this.count = count;
+    }
+
+    String getTag() {
+      return tag;
+    }
+
+    long getCount() {
+      return count;
+    }
+  }
+
+  static class BinsAndCounts {
+    private final List<Double> bins;
+    private final List<Integer> counts;
+
+    BinsAndCounts(List<Double> bins, List<Integer> counts) {
+      this.bins = bins;
+      this.counts = counts;
+    }
+
+    List<Double> getBins() {
+      return bins;
+    }
+
+    List<Integer> getCounts() {
+      return counts;
+    }
+  }
+
+  static class BucketHistogramDataPoint {
     private final List<Long> bucketCounts;
     private final List<Double> explicitBounds;
     private final List<KeyValue> attributesList;
     private final long timeUnixNano;
+    private final boolean exponential;
```

Comment:
If this variable is a flag for an exponential histogram, can we call it `isExponential`?

Reply:
Done.